File: sol/solution/0001-0100/0001. Two Sum/Solution.ex
defmodule Solution do
    @spec two_sum(nums :: [integer], target :: integer) :: [integer]
    def two_sum(nums, target) do
        find(nums, target, %{}, 0)
    end
    defp find([], _target, _map, _index), do: nil
    defp find([num | rest], target, map, index) do
        case Map.get(map, target - num) do nil -> find(rest, target, Map.put(map, num, index), index + 1)
            comp_index -> [comp_index, index]
        end
    end
end

File: sol/solution/0001-0100/0001. Two Sum/Solution.dart
class Solution {
  List<int> twoSum(List<int> nums, int target) {
    Map<int, int> map = {};
    for (int i = 0; i < nums.length; i++) {
      final int? complementIndex = map[target - nums[i]];
      if (complementIndex != null) {
        return [complementIndex, i];
      }
      map[nums[i]] = i;
    }
    return [];
  }
}